homeassistant/components/sensor/ecobee.py

```diff
@@ -0,0 +1,96 @@
+"""
+homeassistant.components.sensor.ecobee
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ecobee Thermostat Component
+
+This component adds support for Ecobee3 Wireless Thermostats.
+You will need to setup developer access to your thermostat,
+and create and API key on the ecobee website.
+
+The first time you run this component you will see a configuration
+component card in Home Assistant.  This card will contain a PIN code
+that you will need to use to authorize access to your thermostat.  You
+can do this at https://www.ecobee.com/consumerportal/index.html
+Click My Apps, Add application, Enter Pin and click Authorize.
+
+After authorizing the application click the button in the configuration
+card.  Now your thermostat and sensors should shown in home-assistant.
+
+You can use the optional hold_temp parameter to set whether or not holds
+are set indefintely or until the next scheduled event.
+
+ecobee:
+  api_key: asdfasdfasdfasdfasdfaasdfasdfasdfasdf
+  hold_temp: True
+
+"""
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.ecobee import NETWORK
+from homeassistant.const import TEMP_FAHRENHEIT
+import logging
+
+DEPENDENCIES = ['ecobee']
+
+SENSOR_TYPES = {
+    'temperature': ['Temperature', TEMP_FAHRENHEIT],
+    'humidity': ['Humidity', '%'],
+    'occupancy': ['Occupancy', '']
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+ECOBEE_CONFIG_FILE = 'ecobee.conf'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the sensors. """
+    if discovery_info is None:
+        return
+    dev = list()
+    while NETWORK is None:
```

Comment:
Don't do this. This is very very bad. Basically causes 100% CPU for waiting.

The problem that is happening here is that you are actively importing NETWORK when this page loads when it is not initialized yet. Instead, do this:
```python
# At top of page
from homeassistant.components import ecobee

for name, data in ecobee.NETWORK.ecobee.sensors.items():
    …
```